lib/Runtime/PlatformAgnostic/Platform/Linux/DateTime.cpp

```diff
@@ -0,0 +1,166 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+#include "Runtime.h"
+#include <time.h>
+#include <sys/time.h>
+#include "ChakraPlatform.h"
+
+namespace Js
+{
+    #define updatePeriod 1000
+
+    const WCHAR *Utility::GetStandardName(size_t *nameLength) {
+        UpdateTimeZoneInfo();
+        *nameLength = standardZoneNameLength;
+        return standardZoneName;
+    }
+
+    const WCHAR *Utility::GetDaylightName(size_t *nameLength) {
+        // We have an abbreviated standard or daylight name
+        // based on the date and time zone we are in.
+        return GetStandardName(nameLength);
+    }
+
+    void Utility::UpdateTimeZoneInfo()
+    {
+        uint32 tickCount = GetTickCount();
+        if (tickCount - lastTimeZoneUpdateTickCount > updatePeriod)
+        {
+            time_t gtime = time(NULL);
+            struct tm ltime;
+
+            tzset();
+            localtime_r(&gtime, &ltime);
+
+            standardZoneNameLength = strlen(ltime.tm_zone);
+
+        #if defined(WCHAR_IS_CHAR16_T)
+            for(int i = 0; i < standardZoneNameLength; i++) {
+                standardZoneName[i] = (char16_t)ltime.tm_zone[i];
+            }
+        #elif defined(WCHAR_IS_WCHAR_T)
+            mbstowcs( (wchar_t*) standardZoneName, ltime.tm_zone,
+                      sizeof(standardZoneName) / sizeof(WCHAR));
+        #else
+        #error "WCHAR should be either wchar_t or char16_t"
+        #endif
+
+            standardZoneName[standardZoneNameLength] = (WCHAR)0;
+            lastTimeZoneUpdateTickCount = tickCount;
+        }
+    }
+
+    // DateTimeHelper ******
+    #define TM_TO_SYSTIME(SYS, TM)           \
+            SYS->wYear = TM->tm_year + 1900; \
+            SYS->wMonth = TM->tm_mon + 1;    \
+            SYS->wDayOfWeek = TM->tm_wday;   \
+            SYS->wDay = TM->tm_mday;         \
+            SYS->wHour = TM->tm_hour;        \
+            SYS->wMinute = TM->tm_min;       \
+            SYS->wSecond = TM->tm_sec
```

Comment:
SYSTEMTIME expects this field to be between 0-59 but tm allows 60 and 61 to be valid values. Consider truncating here (also +1 to Abhijith's suggestion of using a function instead of a macro)